### PR TITLE
Show warning in dialog when using default encryption key

### PR DESCRIPTION
**Pull request by Douile**
Show a warning when changing to a channel that will be using the default encryption key.

<details><summary>Screenshot</summary>

Screenshot_20220822-162039_Meshtastic

</details>

### CODE REVIEW

**Review by andrekir — COMMENTED** [review]
@Douile Thank you for your contribution!
Good catch. Please check my suggestions including the resulting channel name to the dialog for clarity. 👍

File: app/src/main/java/com/geeksville/mesh/ui/ChannelFragment.kt [hunk 1]
```diff
@@ -304,49 +304,60 @@ class ChannelFragment : ScreenFragment("Channel"), Logging {
...
+
+                    MaterialAlertDialogBuilder(requireContext())
+                        .setTitle(R.string.change_channel)
+                        .setMessage(buildString {
+                            append(getString(R.string.are_you_sure_channel))
+                            if (!shouldUseRandomKey) {
+                                append("\n\n")
+                                append(getString(R.string.warning_default_channel))
+                            }
+                        })
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/ChannelFragment.kt]:
```suggestion
                    val humanName = Channel(newSettings.build(), newLoRaConfig.build()).humanName
                    binding.channelNameEdit.setText(humanName)

                    val message = buildString {
                        append(getString(R.string.are_you_sure_channel))
                        if (!shouldUseRandomKey)
                            append("\n\n" + getString(R.string.warning_default_psk).format(humanName))
                    }

                    MaterialAlertDialogBuilder(requireContext())
                        .setTitle(R.string.change_channel)
                        .setMessage(message)
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -155,4 +155,5 @@
     <string name="message">Message</string>
     <string name="mode_append">Append to message</string>
     <string name="mode_instant">Instantly send</string>
+    <string name="warning_default_channel">As you have not changed the channel name you are using the default encryption key (any Meshtastic user can read your messages).</string>
```
Comment (andrekir) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="warning_default_psk">Empty channel names use the default encryption key (any device on %s can read your messages).</string>
```

**Comment by Douile** [comment]
@andrekir thanks for the quick response. I don't see where you left the suggestions though.

**Review by andrekir — COMMENTED** [review]
@Douile aha, missed a button apparently. :)

**Comment by Douile** [comment]
Good changes :+1:

**Comment by andrekir** [comment]
looks great, thanks 👍

**Pull request merged**